### ISSUE TITLE
no pre_logits

### ISSUE THREAD

**Issue by ZhiyuanChen** [facebookresearch/deit#34]
I found that the model trained does not work well in transfer, I suspect it might because the model in this repo does not come with pre_logits

**Comment by TouvronHugo** [comment]
Hi @ZhiyuanChen,
Thanks for your question.
It's strange since we used the same model pre-trained on ImageNet for transfer learning and we are getting quite good results (see Table 8 in our [paper](https://arxiv.org/pdf/2012.12877v1.pdf)). We will try to quickly add the code for the fine-tuning so it will be easier to reproduce our results in transfer learning ;).
I don't understand very well your question about pre-logits, could you give me more context ?
Best,
Hugo

**Comment by ZhiyuanChen** [comment]
> Hi @ZhiyuanChen,
> 
> Thanks for your question.
> 
> It's strange since we used the same model pre-trained on ImageNet for transfer learning and we are getting quite good results (see Table 8 in our [paper](https://arxiv.org/pdf/2012.12877v1.pdf)). We will try to quickly add the code for the fine-tuning so it will be easier to reproduce our results in transfer learning ;).
> 
> I don't understand very well your question about pre-logits, could you give me more context ?
> 
> Best,
> 
> Hugo

I was pretraining on ImageNet and fine tuning on ImageNet ILSVRC-2012. 

In ViT, the head of the model is composed of a two layer mlp, with tanh non-linearity. The first fc layer is called pre_logits in the original implementation of google. The MLP is being replaced by a single fc layer during fine tuning.

**Comment by TouvronHugo** [comment]
Thank you for the details I better understand your question.
For DeiT there is no pre-training with an MLP head. For training we use a simple linear classifier.
Best,
Hugo
